zebra-state/src/service/finalized_state/zebra_db/transparent.rs

```diff
@@ -235,44 +235,25 @@ impl ZebraDb {
         let tx_loc_by_transparent_addr_loc =
             self.db.cf_handle("tx_loc_by_transparent_addr_loc").unwrap();
 
-        // Manually fetch the entire addresses' transaction locations
-        let mut addr_transactions = BTreeSet::new();
-
         // A potentially invalid key representing the first UTXO send to the address,
         // or the query start height.
-        let mut transaction_location = AddressTransaction::address_iterator_start(
+        let transaction_location = AddressTransaction::address_iterator_start(
             address_location,
             *query_height_range.start(),
         );
 
-        loop {
-            // Seek to a valid entry for this address, or the first entry for the next address
-            transaction_location = match self
-                .db
-                .zs_next_key_value_from(&tx_loc_by_transparent_addr_loc, &transaction_location)
-            {
-                Some((transaction_location, ())) => transaction_location,
-                // We're finished with the final address in the column family
-                None => break,
-            };
-
-            // We found the next address, so we're finished with this address
-            if transaction_location.address_location() != address_location {
-                break;
-            }
-
-            // We're past the end height, so we're finished with this query
-            if transaction_location.transaction_location().height > *query_height_range.end() {
-                break;
-            }
-
-            addr_transactions.insert(transaction_location);
-
-            // A potentially invalid key representing the next possible output
-            transaction_location.address_iterator_next();
-        }
+        let last_transaction_location = AddressTransaction::new(
+            address_location,
+            TransactionLocation::from_usize(*query_height_range.end(), u16::MAX.into()),
+        );
```

Comment:
The constant here should be `MAX_ON_DISK_OUTPUT_INDEX`, any higher indexes are invalid in the database.

Optional: Can you move this limit code into an `AddressTransaction::address_iterator_end()` method?

Unlike `address_iterator_start()`, we can't use the `address_location` to improve performance here. But it's helpful to have all the range logic in the same place.

Reply:
> The constant here should be `MAX_ON_DISK_OUTPUT_INDEX`, any higher indexes are invalid in the database.

`OutputIndex` is in `address_location`, this one is a `u16`: https://github.com/ZcashFoundation/zebra/blob/use-range-iter/zebra-state/src/service/finalized_state/disk_format/block.rs#L70-L87

Reply:
> Optional: Can you move this limit code into an AddressTransaction::address_iterator_end() method?

Yep, moved in https://github.com/ZcashFoundation/zebra/pull/7732/commits/7662b6ba9e27d21bd531565719587938b7bb3337.